frontend/pedigree.tsx

```diff
@@ -0,0 +1,58 @@
+import React, { Component, useRef, useEffect } from 'react'
+import { Network } from 'vis-network';
+import "vis-network/styles/vis-network.css"
+import { useHistory } from "react-router-dom";
+
+
+const PedigreeNetwork = ({pedigree}) => {
+
+    const options = {
+      width: Math.round(window.innerWidth * 0.90) + 'px',
+      height: Math.round(window.innerHeight * 0.80) + 'px',
+      layout: {
+        hierarchical: {
+          enabled: true,
+          direction: "DU",
+          sortMethod: 'directed',
+          //shakeTowards: "roots",
+          levelSeparation: 150,
+          parentCentralization:true,
+          edgeMinimization: true,
+          blockShifting: true,
+          nodeSpacing: 100
+        }
+      },
+      edges: {
+        arrows: { to: { enabled: true, scaleFactor: 0.50 } }
+      },
+      interaction: {
+        navigationButtons: true,
+        keyboard: true
+      },
+      physics: false
+    }
+
+    // A reference to the div rendered by this component
+    const domNode = useRef(null);
+    const network = useRef(null);
+    const history = useHistory();
+
+    useEffect(
+    () => {
+      network.current = new Network(domNode.current, pedigree, options);
+      network.current.on("doubleClick", (params) => {
+          if (params.nodes.length > 0) {
+            const nodeid = params.nodes[0];
+            history.push("/individual/" + nodeid);
+          }
+      });
+    },
+    [domNode, network, pedigree, options]);
```

Comment:
Note that refs from useRef don't ever change so these don't need to be listed in the dependency list, see https://reactjs.org/docs/hooks-reference.html#useref
I think the reason why this still works is because the domNode ref is updated before your callback is run. One alternative would be to put the ref in a state instead, as in https://reactjs.org/docs/hooks-faq.html#how-can-i-measure-a-dom-node

You don't need to list options in the dependency list either since it is a constant reference (declared with `const!)

Reply:
Ok, all these concepts are still quite loose for me. But I am looking forward to properly study react! Nice to have someone to point in the right direction.

Reply:
Seems like you're picking it up very quickly! Feel free to ask questions in the slack, there's the #frontend channel which has mainly been used to discuss React pecularities so far :)

Reply:
How can I see the frontend channel? I need to be added maybe?